templates/writing-task-topics.rst

```diff
@@ -0,0 +1,240 @@
+          types of tasks and pages into this section:
+
+  - Tasks that commonly use this task
+  
+  - Tasks that can be used instead of this task
+
+  - Pages in the Processing and Frameworks sections of the Science Pipelines documentation.
+
+In the case of `ProcessCcd`, we would simply say is called as a
+`command line task`, where for e.g. IsrTask, we would say that that it
+is most commonly called by `ProcessCcd`.
+
+.. _task-topics-config:	  
+    
+Configuration
+=============
+
+.. _task-topics-retarg:
+
+Retargetable Subtasks
+---------------------
+
+.. Note:: This content will be automatically derived from the Task
+          code (in which keywords that describe the subtasks in the
+          python code, including a `doc` keyword in which the subtask
+          is described, will already be written).
+
+.. _task-topics-params:
+   
+Parameters
+----------
+
+
+.. Note:: Like the Retargetable Subtasks, this content will be derived
+          from keywords that describe the parameters where they are
+          defined in the Task Config class.
+
+
+.. _task-topics-python:
+   
+Python usage
+============
+
+.. _task-topics-initzn:
+
+Class initialization
+--------------------
+
+- The interface declaring the instance of the class will be
+  picked up automatically from the interface of the `__init__` method
+  and will not require developer input.
+  
+.. _task-topics-run:
+	  
+Run method
+----------
+
+- Similarly to the Class initialization, the description of the
+  interface for calling the primary entrypoint function of the class
+  will be picked up automatically from the interface of the `run`
+  method and will not require developer input.
+
+  
+.. _task-topics-debug:
+
+Debugging
+=========
+
+- Also similarly to the Class initialization, information on the
+  debugging parameters, specifically their name, type, and
+  description, will be picked up automatically from docstrings in the
+  class definition.
+
+.. _task-topics-examples:
+   
+Examples
+========
+
+.. Note:: Making this can be a substantial job which requires writing
+          an actual example and then going through and describing line
+          by line in comments inside of it what the example is doing.
+
+
+The example should be very stripped down and use only the basic functionality of the Task.
+
+To give some specific guidance, we will give a few pointers for how
+one might write an example for IsrTask, which we will call
+`exampleIsrTask.py`, then describe it with reST.
+
+In this particular case, we need to use some functions which are
+normally in the `utils.py` class, and to make it more transparent, we
+might want to strip this down and rewrite them locally in the
+`exampleUtils.py` class.
+
+Next, we describe some of the details for the content of `exampleIsrTask.py`.
+
+Where needed, when describing code, the python code block directive can be used as so:
+
+.. code-block:: rst
+
+  .. code-block:: python
```

Comment:
Ok - i tried to be careful in all cases (this one is a double indent and requires 6 for the python code lines), and will watch this henceforth.